src/libConsensus/ConsensusLeader.cpp

```diff
@@ -139,11 +139,54 @@ void ConsensusLeader::GenerateConsensusSubsets() {
     subset.commitPoints.emplace_back(m_commitPointMap.at(m_myID));
     subset.commitMap.at(m_myID) = true;
 
-    for (unsigned int j = 0; j < m_numForConsensus - 1; j++) {
-      unsigned int index = peersWhoCommitted.at(j);
-      subset.commitPointMap.at(index) = m_commitPointMap.at(index);
-      subset.commitPoints.emplace_back(m_commitPointMap.at(index));
-      subset.commitMap.at(index) = true;
+    // If DS consensus, then first subset should be of dsguard commits only.
+    // Fill in from rest if commits from dsguards < m_numForConsensus
+    if (m_DS && GUARD_MODE && (i == 0)) {
+      unsigned int subsetPeers = 1;  // myself
+      vector<unsigned int> nondsguardIndexes;
+      for (auto index : peersWhoCommitted) {
+        if (index < Guard::GetInstance().GetNumOfDSGuard()) {
+          subset.commitPointMap.at(index) = m_commitPointMap.at(index);
+          subset.commitPoints.emplace_back(m_commitPointMap.at(index));
+          subset.commitMap.at(index) = true;
+          subsetPeers++;
+          if (subsetPeers == m_numForConsensus) {
+            // got all dsguards commit
+            LOG_GENERAL(INFO, "[SubsetID: " << i << "] Got all "
+                                            << m_numForConsensus
+                                            << " commits from ds-guards");
+            break;
+          }
+        } else {
+          nondsguardIndexes.push_back(index);
+        }
+      }
+
+      // check if we fall short of commits from dsguards
+      if (subsetPeers < m_numForConsensus) {
+        // Add from rest of nondsguards commits
+        LOG_GENERAL(
+            WARNING,
+            "[SubsetID: "
+                << i << "] Did'nt got all commits from ds-guards. Expected = "
```

Comment:
"Did'nt got" -> "Didn't get"

Or I suggest to move this log message to before the `if` statement above and just display something shorter like "[SubsetID: X] Guards = X, Non-guards = X"

Reply:
Wanted to log this explicitly only when number of dsguards commit received are short of expectation. So is added inside if statement.

Reply:
Ok, can keep here, but I think a shorter and more informative message like the one proposed can suffice.  Also `m_numForConsensus` is already logged a few lines earlier anyway, so no need to display it again here.

And if you keep the log here (inside the if), and display the exact number of guards and non-guards, then that next log line below is already implied, so no need for it.

Reply:
ok